src/framework/prefabs/scripts/utils.ts

```diff
@@ -0,0 +1,86 @@
+import {pascalCase} from 'change-case';
+
+const FontWeightName: Record<number, string> = {
+  100: 'Thin',
+  200: 'ExtraLight',
+  300: 'Light',
+  400: 'Regular',
+  500: 'Medium',
+  600: 'SemiBold',
+  700: 'Bold',
+  800: 'ExtraBold',
+  900: 'Black',
+};
+
+const DiezFontStyle: Record<string, string> = {
+  regular: 'FontStyle.Normal',
+  italic: 'FontStyle.Italic',
+  bold: 'FontStyle.Bold',
+};
+
+/**
+ * Abstraction to parse font data from different sources into an internal
+ * standard format.
+ *
+ * @ignore
+ */
+abstract class FontCollectionParser {
+  readonly abstract name: string;
+  readonly abstract instanceConstructor: string;
+  collection = new Map<string, string>();
+
+  protected abstract parseVariation (variant: string): {style: string, weight: number};
+
+  protected addToCollection (family: string, style: string, weight: number) {
+    this.collection.set(
+      pascalCase(`${family}-${FontWeightName[weight]}${weight}-${style === 'regular' ? '' : style}`),
+      `${this.instanceConstructor}('${family}', {weight: ${weight}, style: ${DiezFontStyle[style]}})`,
```

Comment:
👀

Reply:
this is quite hard to read in perspective 🤔

Reply:
It is not the most readable code I've seen from you, but you know my feelings — if it works, let's ship it.  If it becomes a problem, we can revise it in the future.